tests/components/vallox/test_config_flow.py

```diff
@@ -0,0 +1,242 @@
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"host": "already_configured"}
+
+
+async def test_import(hass: HomeAssistant) -> None:
+    """Test that import is handled."""
+    name = "Vallox 90 MV"
+
+    with patch(
+        "vallox_websocket_api.Vallox.get_info",
+        return_value={"model": "Vallox 90 MV"},
+    ), patch(
+        "homeassistant.components.vallox.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={"host": "1.2.3.4", "name": name},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == name
+    assert result["data"] == {"host": "1.2.3.4", "name": "Vallox 90 MV"}
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_invalid_host(hass: HomeAssistant) -> None:
+    """Test that invalid host error is handled during import."""
+    name = "Vallox 90 MV"
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data={"host": "vallox90mv.&host.name", "name": name},
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "invalid_host"
+
+
+async def test_import_already_configured(hass: HomeAssistant) -> None:
+    """Test that an already configured Vallox device is handled during import."""
+    name = "Vallox 145 MV"
+
+    with patch(
+        "homeassistant.components.vallox.config_flow.ConfigFlow.host_already_configured",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={"host": "40.10.20.30", "name": name},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_import_cannot_connect(hass: HomeAssistant) -> None:
+    """Test that cannot connect error is handled."""
+    name = "Vallox 90 MV"
+
+    with patch(
+        "vallox_websocket_api.Vallox.get_info",
+        side_effect=OSError,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={"host": "1.2.3.4", "name": name},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "cannot_connect"
+
+
+async def test_import_unknown_exception(hass: HomeAssistant) -> None:
+    """Test that cannot connect error is handled."""
+    name = "Vallox 245 MV"
+
+    with patch(
+        "homeassistant.components.vallox.config_flow.validate_host",
+        side_effect=Exception,
+    ):
```

Comment:
Patch the PyPI package code instead of the config flow code

Reply:
I guess this needs to be done in the same way as https://github.com/home-assistant/core/pull/62780/files#r775270142?

Reply:
Done.